src/glfw/window.cc

```diff
@@ -0,0 +1,233 @@
+    // clang-format on
+} // setWindowHints
+
+std::string
+getHandleAddressString( const Window* window )
+{
+    return std::to_string( reinterpret_cast<uint64_t>( window ) );
+} // getHandleAddressString
+
+void
+logGLFWaction(
+    const std::string& action,
+    const std::string& handleAddressString = {},
+    const std::string& additional = {} //
+)
+{
+    std::string output = "GLFW " + action + ".";
+
+    if ( !handleAddressString.empty() )
+    {
+        // clang-format off
+        output += "\n"
+                  "Handle = " + handleAddressString;
+        // clang-format on
+
+        if ( !additional.empty() )
+        {
+            output += ", " + additional + ".";
+        }
+    }
+
+    spdlog::info( output );
+} // logGLFWaction
+
+} // namespace
+
+WindowType*
+Window::createWindow(
+    const WindowConfig& config,
+    Window* bound_handle //
+)
+{
+    setWindowHints();
+
+    WindowType* glfwWindow = glfwCreateWindow(
+        config.width,
+        config.height,
+        config.title,
+        config.fullscreen ? glfwGetPrimaryMonitor() : nullptr,
+        nullptr );
+
+    glfwSetWindowUserPointer( glfwWindow, bound_handle );
+    glfwSetFramebufferSizeCallback( glfwWindow, framebufferSizeCallback );
+
+    logGLFWaction(
+        "create window",
+        getHandleAddressString( bound_handle ) //
+    );
+
+    return glfwWindow;
+} // createWindow
+
+void
+Window::destroyWindow( WindowType* glfwWindow )
+{
+    Window* window = getHandle( glfwWindow );
+
+    glfwDestroyWindow( glfwWindow );
+
+    logGLFWaction(
+        "destroy window",
+        getHandleAddressString( window ) //
+    );
+} // destroyWindow
+
+void
+Window::framebufferSizeCallback(
+    WindowType* glfwWindow,
+    int width,
+    int height //
+)
+{
+    Window* window = getHandle( glfwWindow );
+
+    logGLFWaction(
+        "framebuffer resize",
+        getHandleAddressString( window ),
+        "width = " + std::to_string( width ) + ", height = " + std::to_string( height ) //
+    );
+
+    window->m_resize_callback( window, width, height );
+} // framebufferSizeCallback
+
+FramebufferSizeType
+Window::getFramebufferSize() const&
+{
+    int width = 0;
+    int height = 0;
+
+    glfwGetFramebufferSize( m_handle.get(), &width, &height );
+
+    return std::make_pair( width, height );
+} // getFramebufferSize
+
+void
+Window::setWindowed() const&
+{
+    constexpr int xpos = 0;
+    constexpr int ypos = 0;
+
+    glfwSetWindowMonitor(
+        m_handle.get(),
+        nullptr,
+        xpos,
+        ypos,
+        getFramebufferWidth(),
+        getFramebufferHeight(),
+        GLFW_DONT_CARE );
+
+    logGLFWaction(
+        "choose windowed mode",
+        getHandleAddressString( this ) //
+    );
+} // setWindowed
+
+void
+Window::setFullscreen() const&
+{
+    GLFWmonitor* monitor = glfwGetPrimaryMonitor();
+
+    glfwSetWindowMonitor(
+        m_handle.get(),
+        monitor,
+        0,
+        0,
```

Comment:
What do these zeros stand for?

Reply:
Reference: The window position is ignored when setting a monitor.
The best solution is to put GLFW_DONT_CARE here.